code/game/objects/items/weapons/shards.dm

```diff
@@ -25,8 +24,7 @@
 									"<span class='danger'>[user] is slitting [user.p_their()] throat with [src]! It looks like [user.p_theyre()] trying to commit suicide.</span>"))
 		return BRUTELOSS
 
-/obj/item/shard/Initialize(mapload)
-	. = ..()
+/obj/item/shard/proc/initial_icon_state()
 	AddComponent(/datum/component/caltrop, force)
```

Comment:
```suggestion
```
I'd move this one back to `Initialize` as it has little to do with the `icon_state`. You always want this behaviour as well for any subtype of `obj/item/shard`